server/src/test/java/com/aws/greengrass/cli/CLIEventStreamAgentTest.java

```diff
@@ -533,4 +533,33 @@ void test_createDebugPassword() throws IOException {
             assertEquals("ABCDE", response.getCertificateSHA256Hash());
         }
     }
+
+    @Test
+    void test_persist_last_five_successful_local_deployments() throws IOException {
+        try (Context context = new Context()) {
+            Topics cliServiceConfig = Topics.of(context, TEST_SERVICE, null);
+
+            CLIEventStreamAgent.LocalDeploymentDetails localDeploymentDetails = new CLIEventStreamAgent.LocalDeploymentDetails();
+            for (int i = 1; i <= 7; i++) {
+                String deploymentId = "deploymentId" + i;
+                localDeploymentDetails.setDeploymentId(deploymentId);
+                localDeploymentDetails.setDeploymentType(Deployment.DeploymentType.LOCAL);
+                if (i == 3) {
+                    localDeploymentDetails.setStatus(DeploymentStatus.FAILED);
+                } else {
+                    localDeploymentDetails.setStatus(DeploymentStatus.SUCCEEDED);
+                }
+                cliEventStreamAgent.persistLocalDeployment(cliServiceConfig, localDeploymentDetails.convertToMapOfObject());
+                // to ensure topic add as order
+                Thread.sleep(1000);
```

Comment:
this is going to slow things down, just sleep 1ms and use `context.waitForPublishQueueToClear()` after sleeping